pkcs11/src/new/functions/key_management.rs

```diff
@@ -86,14 +87,14 @@ impl Pkcs11 {
                 wrapped_key_bytes,
                 &mut wrapped_key_len,
             )
-        } {
-            CKR_OK => {
-                let len = usize::try_from(wrapped_key_len)
-                    .or(Err(Error::InvalidInput("Wrapped key is too long")))?;
-                Ok(unsafe { Vec::from_raw_parts(wrapped_key_bytes, len, len) })
-            }
-            err => Err(Error::Pkcs11(err)),
-        }
+        })
+        .to_result()?;
+
+        //let len = usize::try_from(wrapped_key_len)
+        //.or(Err(Error::InvalidInput("Wrapped key is too long")))?;
+        //Ok(unsafe { Vec::from_raw_parts(wrapped_key_bytes, len, len) })
```

Comment:
Here I think we should follow section [`5.2`](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/errata01/os/pkcs11-base-v2.40-errata01-os-complete.html#_Toc441755801) and call the function twice to first determine the length of the output buffer and then really call the function to produce output. I did something similar in `sign`.